src/docs/testing/common-errors.md

```diff
@@ -0,0 +1,451 @@
+
+Here is an _incomplete_ list of widgets that expect specific parent widgets within the Flutter Framework. Feel free to submit a PR to add more of such widgets. 
+
+| Widget                            | Expected parent widget(s) |
+| :-------------------------------- | ------------------------: |
+| Flexible                          |      Row, Column, or Flex |
+| Expanded (a specialized Flexible) |      Row, Column, or Flex |
+| Positioned                        |                     Stack |
+| Flexible                          |      Row, Column, or Flex |
+| TableCell                         |                     Table |
+
+
+
+### How to fix it?
+
+The fix should be obvious once you know what parent widget is missing. 
+
+
+## ‘setState called during build’
+
+
+### What does the error look like?
+
+<!-- ![alt_text](images/image6.png "image_tooltip") -->
+
+```
+The following assertion was thrown building DialogPage(dirty, dependencies: [_InheritedTheme, _LocalizationsScope-[GlobalKey#59a8e]], state: _DialogPageState#f121e):
+setState() or markNeedsBuild() called during build.
+
+This Overlay widget cannot be marked as needing to build because the framework is already in the process of building widgets.  A widget can be marked as needing to be built during the build phase only if one of its ancestors is currently building. This exception is allowed because the framework builds parent widgets before children, which means a dirty descendant will always be built. Otherwise, the framework might not visit this widget during this build phase.
+The widget on which setState() or markNeedsBuild() was called was: Overlay-[LabeledGlobalKey<OverlayState>#783d6]
+```
+
+### How might you run into the error?
+
+In general, this error occurs when the `setState` method is called within the build method. 
+
+A common scenario where this error would occur is when attempting to trigger a Dialog from within the build method. This scenario is often motivated by the requirement of showing some information to the user immediately upon their entry to a new page.  
+
+Below is a snippet that seems to be a common culprit of this error:
+
+
+```dart
+Widget build(BuildContext context) {
+
+  // Don't do this. 
+  showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: Text("Alert Dialog"),
+        );
+      });
+
+  return Center(
+    child: Column(
+      children: <Widget>[
+        Text('Show Material Dialog'),
+      ],
+    ),
+  );
+}
+```
+
+
+You can’t see the call to `setState` here, but it’s being used as part of `showDialog`, which is why this error is coming up. The build method is never the right place to call `showDialog`. The `build` method could be called by the framework for every frame, e.g., during an animation. 
```

Comment:
=> You don't see the explicit call to `setState`, but it's called by `showDialog`. The `build` method is not the right place to call `showDialog` because `build` can be called by the framework for every frame, for example, during an animation.

Reply:
Done